src/bussinessLogic/ManageData.java

```diff
@@ -30,7 +30,7 @@ public void managePracticeData() {
         this.manageMap.sortCharMap();
         this.manageArray.setCharArrayFromSortedMap();
         // reset the charMap to store new data
-        this.manageMap.setNewCharMap();
+        this.manageMap = new ManageMap();
```

Comment:
In `managePracticeData` we need to create a new empty map to store the time (otherwise it would add to the time that is already stored and the ranking would therefore become irrelevant). This was done before by calling the `setNewCharMap` method, but since we moved it in the constructor we have to instantiate an object to call it. I'm thinking we can make a method called `resetCharMap` that calls  `setNewCharMap` and thus creating a new empty map without instantiating another object. Or we can move the method outside the constructor and call it protected instead of private (so it can be accessed without the need of constructor, but still limit its access mode to members of the same package). Is there another way to create an empty map by using the same object?

Reply:
Why do you need an empty map?